### DISCUSSION TITLE
Is it possible to show a child span's timestamp instead of span lifetime?

### DISCUSSION THREAD

**Discussion by Patchnote-v2** [SigNoz/signoz#1514] [Q&A]
I'm using OpenTelemetry to track user actions on a webpage.  This is working fine, but the way the information is presented isn't as useful as I'd like it to be.  This is because I don't need to time the child spans, but rather use them as a timestamp of when an even happened.  For example, I have the following basic trace that all say `0 ms`:
![image](https://user-images.githubusercontent.com/11915225/184808385-47ec3918-ba8e-4eee-adf4-8dba6dc84dc9.png)

If I focus on the first span, the data is there (`2386 ms`):
![image](https://user-images.githubusercontent.com/11915225/184808433-171f2646-7cb2-4dd6-8751-dece83293076.png)

I'm new to OTel and SigNoz so I'm probably using OTel in the wrong way anyway and perhaps should be using another library entirely.  Or perhaps SigNoz simply doesn't support what I'm asking and a different product would do what I want?  Some feedback would be much appreciated, if this is the right place to ask!

**Comment by pranay01** [comment]
hey @Patchnote-v2 You are trying this in an interesting way which we have not thought about. Currently, the product is designed around use case where the time duration of the span is more important to the users - so that they can figure out if a step is taking longer time. Hence the time when the event started is not shown prominently in the UI.

I am curious, what are you trying to understand by the start time of the event? why is it important to you?

Also, is this a reactjs/angular app which you have instrumented with SIgNoz?

**Reply by Patchnote-v2** [reply]
Hi, thanks for the reply!

We're not currently using React or Angular.  The main goal is to log important actions the user takes in order to analyze how long users spend on certain parts of the page.  It's a long page that they'll be spending a lot of time on, and so we'd like to get information about particularly popular sections and sections that they spend more time than average on.  My thinking was to have a "session" span for the lifetime of the page with child spans that denote user actions.

It does make sense that the goal of the product is oriented towards profiling, especially since every screenshot and code example that I saw for OTel was to show that off.  Since I'm new to telemetry, I'm still learning concepts and terminology so I might be barking up the wrong tree with OTel and/or SigNoz.  I probably haven't learned the keywords yet to get Google to give something more pertinent to my needs.
